staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/cel_validation.go

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"fmt"
+	"math"
+	"sort"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	structuralschema "k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apiextensions-apiserver/third_party/forked/celopenapi/model"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+// unbounded uses nil to represent an unbounded cardinality value.
+var unbounded *uint64 = nil
+
+// CELSchemaContext keeps track of data used by x-kubernetes-validations rules for a specific schema node.
+type CELSchemaContext struct {
+	// withinValidationRuleScope is true if the schema at the current level or above have x-kubernetes-validations rules. typeInfo
+	// should only be populated for schema nodes where this is true.
+	withinValidationRuleScope bool
+
+	// typeInfo is lazily loaded for schema nodes withinValidationRuleScope and may be
+	// populated one of two possible ways:
+	//   1. Using a typeInfoAccessor to accessTypeInfo it from the parent's type info. This is a cheap operation and should be
```

Comment:
```suggestion
	//   1. Using a typeInfoAccessor to access it from the parent's type info. This is a cheap operation and should be
```

Reply:
Grrr, my IDE has some silly notions around how refactors should be applied. Fixed. (I'll dig into IDE settings as well).